apps/src/templates/census2017/CensusMap.jsx

```diff
@@ -0,0 +1,345 @@
+/* global google adjustScroll*/
+import $ from 'jquery';
+import React, { PropTypes, Component } from 'react';
+import ReactDOM from 'react-dom';
+
+class CensusMapInfoWindow extends Component {
+  static propTypes = {
+    handleTakeSurveyClick: PropTypes.func.isRequired,
+    schoolId: PropTypes.string.isRequired,
+    schoolName: PropTypes.string.isRequired,
+    city: PropTypes.string.isRequired,
+    state: PropTypes.string.isRequired,
+    location: PropTypes.string.isRequired,
+    teachesCs: PropTypes.string.isRequired
+  };
+
+  render() {
+    let censusMessage;
+
+    switch (this.props.teachesCs) {
+      case 'YES':
+        censusMessage = "We believe this school offers Computer Science.";
+        break;
+      case 'NO':
+        censusMessage = "We believe this school offers limited or no Computer Science opportunities.";
+        break;
+      case 'MAYBE':
+        censusMessage = "We have conflicting data for this school.";
+        break;
+      default:
+        censusMessage = "We need data for this school.";
+    }
+
+    const schoolDropdownOption = {
+      value: this.props.schoolId,
+      label: `${this.props.schoolName} - ${this.props.city}, ${this.props.state}`,
+      school: {
+        nces_id: this.props.schoolId,
+        name: this.props.schoolName,
+        city: this.props.city,
+        state: this.props.state,
+        latitude: this.props.location.split(',')[0],
+        longitude: this.props.location.split(',')[1],
+      },
+    };
+
+    return (
+      <div className="census-info-window">
+        <h4>
+          <b>
+            {this.props.schoolName}
+          </b>
+          <br />
+          ({this.props.city}, {this.props.state})
+        </h4>
+        <hr />
+        <div className="census-message">{censusMessage}</div>
+        <div className="button-container">
+          <div className="button-link-div">
+            <a onClick={() => this.props.handleTakeSurveyClick(schoolDropdownOption)}>
+              <div className="button">
+                <div className="button-text">Take the survey for this school</div>
+              </div>
+            </a>
+          </div>
+          <div className="button-link-div">
+            <a href="/yourschool/letter" target="_blank">
+              <div className="button">
+                <div className="button-text">Send the survey to a teacher at this school</div>
+              </div>
+            </a>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+export default class CensusMap extends Component {
+  static propTypes = {
+    onSchoolChange: PropTypes.func.isRequired,
+    fusionTableId: PropTypes.string.isRequired,
+    school: PropTypes.object,
+  };
+
+  gmap;
+  scrollwheelOption = false;
+  draggableOption = false;
+  resizeThrottleTimerId;
+  resizeThrottleTimeoutMs = 500;
+  maxHeightPercentage = 0.65;
+  infoWindow;
+
+  componentWillMount() {
+    this.infoWindow = new google.maps.InfoWindow();
```

Comment:
Is there any cleanup step for this `InfoWindow` component we should do after the `CensusMap` component unmounts?

Reply:
Not sure. If the component unmounts then the reference to the object should be orphaned and it would get garbage collected, right? I'm not clear when the component would unmount.

Reply:
If the reference is orphaned the object will be gc'd, yeah. But since JS lacks destructors, a lot of components require explicit teardown - I don't know the details of `InfoWindow` and figured I'd ask.

Also, might not matter if this component is never unmounted on the page... until you write unit tests :smiling_imp: so makes sense to not worry about it until then.